[2024-06-17T04:10:22.644+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-17T04:10:22.993+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-06-16T00:00:00+00:00 [queued]>
[2024-06-17T04:10:23.065+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-06-16T00:00:00+00:00 [queued]>
[2024-06-17T04:10:23.071+0000] {taskinstance.py:2306} INFO - Starting attempt 6 of 6
[2024-06-17T04:10:23.148+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): avg_emission_by_year> on 2024-06-16 00:00:00+00:00
[2024-06-17T04:10:23.164+0000] {standard_task_runner.py:63} INFO - Started process 904 to run task
[2024-06-17T04:10:23.177+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'avg_emission_by_year', 'scheduled__2024-06-16T00:00:00+00:00', '--job-id', '143', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpos_nd7j7']
[2024-06-17T04:10:23.192+0000] {standard_task_runner.py:91} INFO - Job 143: Subtask avg_emission_by_year
[2024-06-17T04:10:23.585+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year scheduled__2024-06-16T00:00:00+00:00 [running]> on host fa4d239664b0
[2024-06-17T04:10:23.901+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='avg_emission_by_year' AIRFLOW_CTX_EXECUTION_DATE='2024-06-16T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-16T00:00:00+00:00'
[2024-06-17T04:10:23.904+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-17T04:10:23.905+0000] {summary_analysis.py:33} INFO - starting average pollutants emission calculation
[2024-06-17T04:10:23.908+0000] {summary_analysis.py:45} ERROR - error calculating average emission
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 36, in avg_emission_by_year
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 791, in read_json
    json_reader = JsonReader(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 904, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 944, in _get_data_from_filepath
    self.handles = get_handle(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 728, in get_handle
    ioargs = _get_filepath_or_buffer(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 472, in _get_filepath_or_buffer
    raise ValueError(msg)
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
[2024-06-17T04:10:23.911+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-17T04:10:23.912+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 36, in avg_emission_by_year
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 791, in read_json
    json_reader = JsonReader(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 904, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 944, in _get_data_from_filepath
    self.handles = get_handle(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 728, in get_handle
    ioargs = _get_filepath_or_buffer(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 472, in _get_filepath_or_buffer
    raise ValueError(msg)
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
[2024-06-17T04:10:23.928+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=Air_pollution_data_pipeline, task_id=avg_emission_by_year, run_id=scheduled__2024-06-16T00:00:00+00:00, execution_date=20240616T000000, start_date=20240617T041022, end_date=20240617T041023
[2024-06-17T04:10:23.957+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 143 for task avg_emission_by_year (Invalid file path or buffer object type: <class 'NoneType'>; 904)
[2024-06-17T04:10:24.021+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-17T04:10:24.104+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-17T04:10:24.116+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
